inference-engine/src/mkldnn_plugin/nodes/extract_image_patches.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "base.hpp"
+#include "caseless.hpp"
+#include "ie_parallel.hpp"
+#include <vector>
+
+namespace InferenceEngine {
+namespace Extensions {
+namespace Cpu {
+
+struct jit_extract_image_patches_params {
+    size_t IW;
+    size_t OH, OW;
+    size_t KH, KW;
+    size_t SH, SW;
+    size_t dtype_size;
+    size_t block_size;
+    bool need_padding;
+};
+
+struct jit_extract_image_patches_args {
+    uint64_t h_lo_pad;
+    uint64_t h_hi_pad;
+    uint64_t w_lo_pad;
+    uint64_t w_hi_pad;
+    const void* src;
+    void* dst;
+};
+
+struct jit_uni_extract_image_patches_kernel {
+    void (*ker_)(const jit_extract_image_patches_args *);
+    void operator()(const jit_extract_image_patches_args *args) { assert(ker_); ker_(args); }
+    jit_extract_image_patches_params jpp;
+    virtual void create_ker() = 0;
+    explicit jit_uni_extract_image_patches_kernel(jit_extract_image_patches_params jpp) : ker_(nullptr), jpp(jpp) {}
+    virtual ~jit_uni_extract_image_patches_kernel() {}
+};
+
+using details::CaselessEq;
+
+class ExtractImagePatchesImpl : public ExtLayerBase {
+public:
+    explicit ExtractImagePatchesImpl(const CNNLayer*);
+    StatusCode execute(std::vector<Blob::Ptr>&, std::vector<Blob::Ptr>&, ResponseDesc*) noexcept override;
+
+private:
+    std::vector<size_t> _ksizes;
+    std::vector<size_t> _strides;
+    std::vector<size_t> _rates;
+    size_t _pad_left;
+    size_t _pad_top;
+    std::shared_ptr<jit_uni_extract_image_patches_kernel> extract_image_patches_kernel;
+    static const std::set<size_t> _supported_precisions_sizes;
+};
+
+const std::set<size_t> ExtractImagePatchesImpl::_supported_precisions_sizes = {1, 2, 4};
```

Comment:
Seems like it is better to move initialization into cpp

Reply:
Ok, moved to the cpp